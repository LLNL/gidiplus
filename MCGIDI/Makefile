SHELL = /bin/ksh

# <<BEGIN-copyright>>
# Copyright 2019, Lawrence Livermore National Security, LLC.
# See the top-level COPYRIGHT file for details.
# 
# SPDX-License-Identifier: MIT
# <<END-copyright>>

DIRS = Src include lib bin Doc

POPs_PATH = $(abspath ../PoPI)
SMR_PATH = $(abspath ../statusMessageReporting)
NF_PATH = $(abspath ../numericalFunctions)

.PHONY: default include lib bin clean realclean

default: include lib
	cd Test; $(MAKE) default
	$(MAKE) bin

include:
	cd include; $(MAKE)

lib:
	cd lib; $(MAKE) CXX=$(CXX) CXXFLAGS='$(CXXFLAGS)' POPs_PATH='$(POPs_PATH)' SMR_PATH='$(SMR_PATH)' NF_PATH='$(NF_PATH)'

bin:
	cd bin; $(MAKE) CXX=$(CXX) CXXFLAGS='$(CXXFLAGS)' POPs_PATH='$(POPs_PATH)' SMR_PATH='$(SMR_PATH)' NF_PATH='$(NF_PATH)' bin

check:
	cd Test; $(MAKE) CXX=$(CXX) CXXFLAGS='$(CXXFLAGS)' POPs_PATH='$(POPs_PATH)' SMR_PATH='$(SMR_PATH)' NF_PATH='$(NF_PATH)' check

speeds:
	cd Speeds; $(MAKE) CXX=$(CXX) CXXFLAGS='$(CXXFLAGS)' POPs_PATH='$(POPs_PATH)' SMR_PATH='$(SMR_PATH)' NF_PATH='$(NF_PATH)' speeds

clean:
	$(MAKE) doDIRS _DIRS="$(DIRS)" _TARGET=clean
	cd Test; $(MAKE) clean
	cd Speeds; $(MAKE) clean

realclean:
	$(MAKE) doDIRS _DIRS="$(DIRS)" _TARGET=realclean
	cd Test; $(MAKE) realclean
	cd Speeds; $(MAKE) realclean

docs:
	doxygen Doc/doxygen/Doxyfile

doDIRS:
	SAVED_PWD=`pwd`; \
	for DIR in $(_DIRS); do cd $$DIR; $(MAKE) $(_TARGET); cd $$SAVED_PWD; done
