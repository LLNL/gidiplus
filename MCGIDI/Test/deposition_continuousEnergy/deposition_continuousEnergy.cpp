/*
# <<BEGIN-copyright>>
# Copyright 2019, Lawrence Livermore National Security, LLC.
# See the top-level COPYRIGHT file for details.
# 
# SPDX-License-Identifier: MIT
# <<END-copyright>>
*/

#include <stdlib.h>
#include <iostream>
#include <iomanip>
#include <set>

#include "MCGIDI.hpp"

#include "MCGIDI_testUtilities.hpp"

static char const *description = "Loops over temperature and energy, printing the total cross section. If projectile is a photon, see options *-pa* and *-pn*.";

void main2( int argc, char **argv );
/*
=========================================================
*/
int main( int argc, char **argv ) {

    try {
        main2( argc, argv );
        exit( EXIT_SUCCESS ); }
    catch (std::exception &exception) {
        std::cerr << exception.what( ) << std::endl; }
    catch (char const *str) {
        std::cout << str << std::endl; }
    catch (std::string &str) {
        std::cout << str << std::endl;
    }
}
/*
=========================================================
*/
void main2( int argc, char **argv ) {

    PoPI::Database pops( "../../../GIDI/Test/pops.xml" );
    GIDI::Protare *protare;
    GIDI::Transporting::Particles particles;
    std::set<int> reactionsToExclude;
    GIDI::Construction::PhotoMode photo_mode = GIDI::Construction::PhotoMode::nuclearOnly;
    int neutronIndex = pops[PoPI::IDs::neutron];
    int photonIndex = pops[PoPI::IDs::photon];

    std::cerr << "    " << __FILE__;
    for( int i1 = 1; i1 < argc; i1++ ) std::cerr << " " << argv[i1];
    std::cerr << std::endl;

    argvOptions2 argv_options( "crossSections", description );

    argv_options.add( argvOption2( "--map", true, "The map file to use." ) );
    argv_options.add( argvOption2( "--pid", true, "The PoPs id of the projectile." ) );
    argv_options.add( argvOption2( "--tid", true, "The PoPs id of the target." ) );
    argv_options.add( argvOption2( "--pa", false, "Include photo-atomic protare if relevant. If present, disables photo-nuclear unless *-n* present." ) );
    argv_options.add( argvOption2( "--pn", false, "Include photo-nuclear protare if relevant. This is the default unless *-a* present." ) );
    argv_options.add( argvOption2( "-n", false, "If present, add neutron as transporting particle." ) );
    argv_options.add( argvOption2( "-p", false, "If present, add photon as transporting particle." ) );

    argv_options.parseArgv( argc, argv );

    std::string mapFilename = argv_options.find( "--map" )->zeroOrOneOption( argv, "../../../GIDI/Test/all3T.map" );
    std::string projectileID = argv_options.find( "--pid" )->zeroOrOneOption( argv, PoPI::IDs::neutron );
    std::string targetID = argv_options.find( "--tid" )->zeroOrOneOption( argv, "O16" );

    if( argv_options.find( "--pa" )->present( ) ) {
        photo_mode = GIDI::Construction::PhotoMode::atomicOnly;
        if( argv_options.find( "--pn" )->present( ) ) photo_mode = GIDI::Construction::PhotoMode::nuclearAndAtomic;
    }

    GIDI::Map::Map map( mapFilename, pops );
    GIDI::Construction::Settings construction( GIDI::Construction::ParseMode::all, photo_mode );
    protare = map.protare( construction, pops, projectileID, targetID );

    GIDI::Styles::TemperatureInfos temperatures = protare->temperatures( );
    std::string label( temperatures[0].griddedCrossSection( ) );

    if( ( projectileID == PoPI::IDs::neutron ) || argv_options.find( "-n" )->present( ) ) {
        GIDI::Transporting::Particle neutron( PoPI::IDs::neutron );
        particles.add( neutron );
    }

    if( ( projectileID == PoPI::IDs::photon ) || argv_options.find( "-p" )->present( ) ) {
        GIDI::Transporting::Particle photon( PoPI::IDs::photon );
        particles.add( photon );
    }

    MCGIDI::Transporting::MC MC( pops, projectileID, &protare->styles( ), label, GIDI::Transporting::DelayedNeutrons::on, 20.0 );
    MC.crossSectionLookupMode( MCGIDI::Transporting::LookupMode::Data1d::continuousEnergy );
    MCGIDI::DomainHash domainHash( 4000, 1e-8, 10 );
    MCGIDI::Protare *MCProtare;
    MCProtare = MCGIDI::protareFromGIDIProtare( *protare, pops, MC, particles, domainHash, temperatures, reactionsToExclude );

    MCGIDI::Vector<MCGIDI::Protare *> protares( 1 );
    protares[0] = MCProtare;

    for( MCGIDI_VectorSizeType i1 = 0; i1 < (MCGIDI_VectorSizeType) MCProtare->numberOfReactions( ); ++i1 ) {
        MCGIDI::Reaction const &reaction = *MCProtare->reaction( i1 );

        std::cout << std::setw( 40 ) << reaction.label( ).c_str( ) << "  threshold = " << std::setw( 12 ) << reaction.crossSectionThreshold( ) <<
                "  threshold = " << std::setw( 12 ) << reaction.crossSectionThreshold( ) << std::endl;
    }

    for( MCGIDI_VectorSizeType i1 = 0; i1 < (MCGIDI_VectorSizeType) MCProtare->numberOfReactions( ); ++i1 ) {
        MCGIDI::Reaction const &reaction = *MCProtare->reaction( i1 );

        std::cout << "    reaction: " << reaction.label( ).c_str( ) << std::endl;
    }

    for( double temperature = 1e-8; temperature < 2e-3; temperature *= 100.0 ) {
        std::cout << "temperature = " << doubleToString2( "%8.1e", temperature ) << "                       deposition energy  deposition momentum  production energy";
        if( particles.hasParticle( PoPI::IDs::neutron ) ) std::cout << "      neutron gain";
        if( particles.hasParticle( PoPI::IDs::photon ) ) std::cout << "         photon gain";
        std::cout << std::endl;
        for( double energy = 1e-12; energy < 45.0; energy *= 2.0 ) {
            int hashIndex = domainHash.index( energy );

            std::cout << "    energy = " << std::setw( 16 ) << energy << " index = " << std::setw( 6 ) << hashIndex;
            std::cout << doubleToString2( " %16.8e",    MCProtare->depositionEnergy( hashIndex, temperature, energy ) );
            std::cout << doubleToString2( "    %16.8e", MCProtare->depositionMomentum( hashIndex, temperature, energy ) );
            std::cout << doubleToString2( "    %16.8e", MCProtare->productionEnergy( hashIndex, temperature, energy ) );
            if( particles.hasParticle( PoPI::IDs::neutron ) ) std::cout << doubleToString2( "    %16.8e", MCProtare->gain( hashIndex, temperature, energy, neutronIndex ) );
            if( particles.hasParticle( PoPI::IDs::photon ) ) std::cout << doubleToString2( "    %16.8e", MCProtare->gain( hashIndex, temperature, energy, photonIndex ) );
            std::cout << std::endl;
        }
    }

    delete protare;

    delete MCProtare;
}
