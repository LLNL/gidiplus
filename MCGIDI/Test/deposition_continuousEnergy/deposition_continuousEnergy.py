# <<BEGIN-copyright>>
# Copyright 2019, Lawrence Livermore National Security, LLC.
# See the top-level COPYRIGHT file for details.
# 
# SPDX-License-Identifier: MIT
# <<END-copyright>>

import sys
import os

from argparse import ArgumentParser

code = 'deposition_continuousEnergy'

description = """Runs %s on a list of targets for defined options.""" % code

parser = ArgumentParser( description = description )
parser.add_argument( 'targets', action = 'append',                          help = 'List of target IDs to check.' )
parser.add_argument( '-d', action='store_true',                             help = 'If present, pass the -d option to %s.' % code )
parser.add_argument( '--useSlowerContinuousEnergyConversion', action='store_true',  help = 'If present, pass on to %s.' % code )
parser.add_argument( '-v', '--verbose', action = 'count', default = 0,      help = 'Enable verbose output.' )

args = parser.parse_args( )

def checkOptions( target, options ) :

    if( args.verbose > 0 ) : print( )
    output = '%s.%s' % ( code, target )

    if args.d:
        options.append('-d')
    if args.useSlowerContinuousEnergyConversion:
        options.append('--useSlowerContinuousEnergyConversion')
    for option in options : output += '.%s' % option

    benchmarks = os.path.join( 'Benchmarks', output + '.out' ).replace( '..', '.' )
    output = os.path.join( 'Outputs', output + '.out' ).replace( '..', '.' )
    cmd = './%s %s --tid %s > %s' % ( code, ' '.join( options ), target, output )
    if( args.verbose > 0 ) : print( cmd )
    os.system( cmd )
    cmd = '%s diff.py %s/%s %s %s' % (sys.executable, code, code, benchmarks, output)
    if( args.verbose > 0 ) : print( cmd )
    os.system( cmd )

def checkTarget( target ) :

    for options in [ [ '' ], [ '-p' ] ] : checkOptions( target, options )

if( not( os.path.exists( 'Outputs' ) ) ) : os.mkdir( 'Outputs' )
for target in args.targets : checkTarget( target )
