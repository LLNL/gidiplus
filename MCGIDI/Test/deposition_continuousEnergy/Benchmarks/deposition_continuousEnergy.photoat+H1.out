                   H + photon [coherent]  threshold =            0  threshold =            0
                 H + photon [incoherent]  threshold =            0  threshold =            0
e- + e-_anti + H [pair production: electron field]  threshold =        2.044  threshold =        2.044
e- + e-_anti + H [pair production: nuclear field]  threshold =        1.022  threshold =        1.022
                           e- + H{1s1/2}  threshold =            0  threshold =            0
List of reactions:
    reaction: H + photon [coherent]
    reaction: H + photon [incoherent]
    reaction: e- + e-_anti + H [pair production: electron field]
    reaction: e- + e-_anti + H [pair production: nuclear field]
    reaction: e- + H{1s1/2}
temperature =  1.0e-08                          cross section deposition energy  deposition momentum  production energy         photon gain
    energy =            1e-12 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =            2e-12 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =            4e-12 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =            8e-12 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =          1.6e-11 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =          3.2e-11 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =          6.4e-11 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =         1.28e-10 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =         2.56e-10 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =         5.12e-10 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =        1.024e-09 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =        2.048e-09 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =        4.096e-09 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =        8.192e-09 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =       1.6384e-08 index =     96   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =       3.2768e-08 index =    230   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =       6.5536e-08 index =    363   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =      1.31072e-07 index =    497   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =      2.62144e-07 index =    631   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =      5.24288e-07 index =    765   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =      1.04858e-06 index =    899   1.19955430e-05   3.19639509e-19      1.24903853e-11      0.00000000e+00      0.00000000e+00
    energy =      2.09715e-06 index =   1032   2.07863309e-04   5.12729306e-18      4.37303932e-10      0.00000000e+00      0.00000000e+00
    energy =       4.1943e-06 index =   1166   4.49548179e-03   8.14247272e-17      1.88602143e-08      0.00000000e+00      0.00000000e+00
    energy =      8.38861e-06 index =   1300   5.28679107e-01   1.30670506e-15      4.43671902e-06      0.00000000e+00      0.00000000e+00
    energy =      1.67772e-05 index =   1434   3.58517061e+06   6.01411534e+01      6.01411776e+01      0.00000000e+00     -3.58516916e+06
    energy =      3.35544e-05 index =   1567   5.10300397e+05   1.71215542e+01      1.71215863e+01      0.00000000e+00     -5.10299439e+05
    energy =      6.71089e-05 index =   1701   6.54322993e+04   4.39071964e+00      4.39077143e+00      0.00000000e+00     -6.54315271e+04
    energy =      0.000134218 index =   1835   7.73640679e+03   1.03818489e+00      1.03827883e+00      0.00000000e+00     -7.73570519e+03
    energy =      0.000268435 index =   1969   8.56352836e+02   2.29657167e-01      2.29836955e-01      0.00000000e+00     -8.55676258e+02
    energy =      0.000536871 index =   2103   9.03565059e+01   4.81450535e-02      4.84897751e-02      0.00000000e+00     -8.96879297e+01
    energy =       0.00107374 index =   2236   9.66131395e+00   9.65865324e-03      1.02713861e-02      0.00000000e+00     -8.99552456e+00
    energy =       0.00214748 index =   2370   1.53277378e+00   1.87735413e-03      2.71760521e-03      0.00000000e+00     -8.72748517e-01
    energy =       0.00429497 index =   2504   7.35253662e-01   3.74828213e-04      1.13078711e-03      0.00000000e+00     -8.24904824e-02
    energy =       0.00858993 index =   2638   6.46619479e-01   1.54305944e-04      1.10727793e-03      0.00000000e+00     -7.68782867e-03
    energy =        0.0171799 index =   2772   6.24037968e-01   3.49440875e-04      8.74164843e-03      0.00000000e+00     -7.06467701e-04
    energy =        0.0343597 index =   2905   5.88499226e-01   1.19407819e-03     -9.86549547e-03      0.00000000e+00     -6.45930098e-05
    energy =        0.0687195 index =   3039   5.32047101e-01   3.81780772e-03     -1.38712822e-02      0.00000000e+00     -5.93616768e-06
    energy =         0.137439 index =   3173   4.54010615e-01   1.07166689e-02     -1.80898541e-02      0.00000000e+00     -5.60397575e-07
    energy =         0.274878 index =   3307   3.64406113e-01   2.58743864e-02     -1.80980007e-02      0.00000000e+00     -5.80160250e-08
    energy =         0.549756 index =   3441   2.77487305e-01   5.42381849e-02     -7.93546245e-03      0.00000000e+00     -7.71815039e-09
    energy =          1.09951 index =   3574   2.01274244e-01   1.00370813e-01      2.00526334e-02      0.00000000e+00      4.32701318e-07
    energy =          2.19902 index =   3708   1.38657458e-01   1.65356881e-01      7.24009932e-02      0.00000000e+00      2.38807441e-04
    energy =          4.39805 index =   3842   9.12779274e-02   2.48129067e-01      1.51468677e-01      0.00000000e+00      1.15893389e-03
    energy =          8.79609 index =   3976   5.88125411e-02   3.54267699e-01      2.61737207e-01      0.00000000e+00      2.91241144e-03
    energy =          17.5922 index =   4001   3.85619052e-02   5.08403616e-01      4.24540554e-01      0.00000000e+00      5.19189709e-03
    energy =          35.1844 index =   4001   2.70768522e-02   7.77493980e-01      7.04123527e-01      0.00000000e+00      7.71988959e-03
temperature =  1.0e-06                          cross section deposition energy  deposition momentum  production energy         photon gain
    energy =            1e-12 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =            2e-12 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =            4e-12 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =            8e-12 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =          1.6e-11 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =          3.2e-11 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =          6.4e-11 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =         1.28e-10 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =         2.56e-10 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =         5.12e-10 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =        1.024e-09 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =        2.048e-09 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =        4.096e-09 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =        8.192e-09 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =       1.6384e-08 index =     96   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =       3.2768e-08 index =    230   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =       6.5536e-08 index =    363   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =      1.31072e-07 index =    497   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =      2.62144e-07 index =    631   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =      5.24288e-07 index =    765   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =      1.04858e-06 index =    899   1.19955430e-05   3.19639509e-19      1.24903853e-11      0.00000000e+00      0.00000000e+00
    energy =      2.09715e-06 index =   1032   2.07863309e-04   5.12729306e-18      4.37303932e-10      0.00000000e+00      0.00000000e+00
    energy =       4.1943e-06 index =   1166   4.49548179e-03   8.14247272e-17      1.88602143e-08      0.00000000e+00      0.00000000e+00
    energy =      8.38861e-06 index =   1300   5.28679107e-01   1.30670506e-15      4.43671902e-06      0.00000000e+00      0.00000000e+00
    energy =      1.67772e-05 index =   1434   3.58517061e+06   6.01411534e+01      6.01411776e+01      0.00000000e+00     -3.58516916e+06
    energy =      3.35544e-05 index =   1567   5.10300397e+05   1.71215542e+01      1.71215863e+01      0.00000000e+00     -5.10299439e+05
    energy =      6.71089e-05 index =   1701   6.54322993e+04   4.39071964e+00      4.39077143e+00      0.00000000e+00     -6.54315271e+04
    energy =      0.000134218 index =   1835   7.73640679e+03   1.03818489e+00      1.03827883e+00      0.00000000e+00     -7.73570519e+03
    energy =      0.000268435 index =   1969   8.56352836e+02   2.29657167e-01      2.29836955e-01      0.00000000e+00     -8.55676258e+02
    energy =      0.000536871 index =   2103   9.03565059e+01   4.81450535e-02      4.84897751e-02      0.00000000e+00     -8.96879297e+01
    energy =       0.00107374 index =   2236   9.66131395e+00   9.65865324e-03      1.02713861e-02      0.00000000e+00     -8.99552456e+00
    energy =       0.00214748 index =   2370   1.53277378e+00   1.87735413e-03      2.71760521e-03      0.00000000e+00     -8.72748517e-01
    energy =       0.00429497 index =   2504   7.35253662e-01   3.74828213e-04      1.13078711e-03      0.00000000e+00     -8.24904824e-02
    energy =       0.00858993 index =   2638   6.46619479e-01   1.54305944e-04      1.10727793e-03      0.00000000e+00     -7.68782867e-03
    energy =        0.0171799 index =   2772   6.24037968e-01   3.49440875e-04      8.74164843e-03      0.00000000e+00     -7.06467701e-04
    energy =        0.0343597 index =   2905   5.88499226e-01   1.19407819e-03     -9.86549547e-03      0.00000000e+00     -6.45930098e-05
    energy =        0.0687195 index =   3039   5.32047101e-01   3.81780772e-03     -1.38712822e-02      0.00000000e+00     -5.93616768e-06
    energy =         0.137439 index =   3173   4.54010615e-01   1.07166689e-02     -1.80898541e-02      0.00000000e+00     -5.60397575e-07
    energy =         0.274878 index =   3307   3.64406113e-01   2.58743864e-02     -1.80980007e-02      0.00000000e+00     -5.80160250e-08
    energy =         0.549756 index =   3441   2.77487305e-01   5.42381849e-02     -7.93546245e-03      0.00000000e+00     -7.71815039e-09
    energy =          1.09951 index =   3574   2.01274244e-01   1.00370813e-01      2.00526334e-02      0.00000000e+00      4.32701318e-07
    energy =          2.19902 index =   3708   1.38657458e-01   1.65356881e-01      7.24009932e-02      0.00000000e+00      2.38807441e-04
    energy =          4.39805 index =   3842   9.12779274e-02   2.48129067e-01      1.51468677e-01      0.00000000e+00      1.15893389e-03
    energy =          8.79609 index =   3976   5.88125411e-02   3.54267699e-01      2.61737207e-01      0.00000000e+00      2.91241144e-03
    energy =          17.5922 index =   4001   3.85619052e-02   5.08403616e-01      4.24540554e-01      0.00000000e+00      5.19189709e-03
    energy =          35.1844 index =   4001   2.70768522e-02   7.77493980e-01      7.04123527e-01      0.00000000e+00      7.71988959e-03
temperature =  1.0e-04                          cross section deposition energy  deposition momentum  production energy         photon gain
    energy =            1e-12 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =            2e-12 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =            4e-12 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =            8e-12 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =          1.6e-11 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =          3.2e-11 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =          6.4e-11 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =         1.28e-10 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =         2.56e-10 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =         5.12e-10 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =        1.024e-09 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =        2.048e-09 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =        4.096e-09 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =        8.192e-09 index =      0   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =       1.6384e-08 index =     96   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =       3.2768e-08 index =    230   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =       6.5536e-08 index =    363   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =      1.31072e-07 index =    497   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =      2.62144e-07 index =    631   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =      5.24288e-07 index =    765   9.98317600e-06   2.61980381e-19      9.88755337e-12      0.00000000e+00      0.00000000e+00
    energy =      1.04858e-06 index =    899   1.19955430e-05   3.19639509e-19      1.24903853e-11      0.00000000e+00      0.00000000e+00
    energy =      2.09715e-06 index =   1032   2.07863309e-04   5.12729306e-18      4.37303932e-10      0.00000000e+00      0.00000000e+00
    energy =       4.1943e-06 index =   1166   4.49548179e-03   8.14247272e-17      1.88602143e-08      0.00000000e+00      0.00000000e+00
    energy =      8.38861e-06 index =   1300   5.28679107e-01   1.30670506e-15      4.43671902e-06      0.00000000e+00      0.00000000e+00
    energy =      1.67772e-05 index =   1434   3.58517061e+06   6.01411534e+01      6.01411776e+01      0.00000000e+00     -3.58516916e+06
    energy =      3.35544e-05 index =   1567   5.10300397e+05   1.71215542e+01      1.71215863e+01      0.00000000e+00     -5.10299439e+05
    energy =      6.71089e-05 index =   1701   6.54322993e+04   4.39071964e+00      4.39077143e+00      0.00000000e+00     -6.54315271e+04
    energy =      0.000134218 index =   1835   7.73640679e+03   1.03818489e+00      1.03827883e+00      0.00000000e+00     -7.73570519e+03
    energy =      0.000268435 index =   1969   8.56352836e+02   2.29657167e-01      2.29836955e-01      0.00000000e+00     -8.55676258e+02
    energy =      0.000536871 index =   2103   9.03565059e+01   4.81450535e-02      4.84897751e-02      0.00000000e+00     -8.96879297e+01
    energy =       0.00107374 index =   2236   9.66131395e+00   9.65865324e-03      1.02713861e-02      0.00000000e+00     -8.99552456e+00
    energy =       0.00214748 index =   2370   1.53277378e+00   1.87735413e-03      2.71760521e-03      0.00000000e+00     -8.72748517e-01
    energy =       0.00429497 index =   2504   7.35253662e-01   3.74828213e-04      1.13078711e-03      0.00000000e+00     -8.24904824e-02
    energy =       0.00858993 index =   2638   6.46619479e-01   1.54305944e-04      1.10727793e-03      0.00000000e+00     -7.68782867e-03
    energy =        0.0171799 index =   2772   6.24037968e-01   3.49440875e-04      8.74164843e-03      0.00000000e+00     -7.06467701e-04
    energy =        0.0343597 index =   2905   5.88499226e-01   1.19407819e-03     -9.86549547e-03      0.00000000e+00     -6.45930098e-05
    energy =        0.0687195 index =   3039   5.32047101e-01   3.81780772e-03     -1.38712822e-02      0.00000000e+00     -5.93616768e-06
    energy =         0.137439 index =   3173   4.54010615e-01   1.07166689e-02     -1.80898541e-02      0.00000000e+00     -5.60397575e-07
    energy =         0.274878 index =   3307   3.64406113e-01   2.58743864e-02     -1.80980007e-02      0.00000000e+00     -5.80160250e-08
    energy =         0.549756 index =   3441   2.77487305e-01   5.42381849e-02     -7.93546245e-03      0.00000000e+00     -7.71815039e-09
    energy =          1.09951 index =   3574   2.01274244e-01   1.00370813e-01      2.00526334e-02      0.00000000e+00      4.32701318e-07
    energy =          2.19902 index =   3708   1.38657458e-01   1.65356881e-01      7.24009932e-02      0.00000000e+00      2.38807441e-04
    energy =          4.39805 index =   3842   9.12779274e-02   2.48129067e-01      1.51468677e-01      0.00000000e+00      1.15893389e-03
    energy =          8.79609 index =   3976   5.88125411e-02   3.54267699e-01      2.61737207e-01      0.00000000e+00      2.91241144e-03
    energy =          17.5922 index =   4001   3.85619052e-02   5.08403616e-01      4.24540554e-01      0.00000000e+00      5.19189709e-03
    energy =          35.1844 index =   4001   2.70768522e-02   7.77493980e-01      7.04123527e-01      0.00000000e+00      7.71988959e-03
